### ISSUE TITLE
Auto3dSeg GPU Customization Test Non-Determinsistically Failing

### ISSUE THREAD

**Issue by peterhessey** [Project-MONAI/MONAI#6110]
**Describe the bug**
When running the test `python -m tests.test_integration_gpu_customization` it sometimes fails.

**To Reproduce**
1. Go to head of MONAI repo
2. Create and activate conda env using `conda env create --file environment-dev.yml && conda activate monai`
3. Run commands `python -m tests.test_integration_gpu_customization`

**Expected behavior**
Test passes every time.

**Actual behaviour**

Test occasionally fails (when DINTS is selected as best algorithm for ensembling, I believe) with following outputs:

```shell

======================================================================
ERROR: test_ensemble_gpu_customization (__main__.TestEnsembleGpuCustomization)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/azureuser/localfiles/Repos/MONAI/tests/test_integration_gpu_customization.py", line 149, in test_ensemble_gpu_customization
    preds = ensemble(pred_param)
  File "/home/azureuser/localfiles/Repos/MONAI/monai/apps/auto3dseg/ensemble_builder.py", line 164, in __call__
    pred = infer_instance.predict(predict_files=[file], predict_params=param)
  File "/home/azureuser/localfiles/Repos/MONAI/monai/apps/auto3dseg/bundle_gen.py", line 271, in predict
    inferer = self.get_inferer(**params)
  File "/home/azureuser/localfiles/Repos/MONAI/monai/apps/auto3dseg/bundle_gen.py", line 255, in get_inferer
    return infer_class.InferClass(configs_path, *args, **kwargs)
  File "/tmp/tmprkmn_prz/workdir/dints_0/scripts/infer.py", line 84, in __init__
  File "/anaconda/envs/monai/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for DiNTS:
        Missing key(s) in state_dict:
        ...
        Unexpected key(s) in state_dict:
        ...
```

**Screenshots**

<img width="931" alt="image" src="https://user-images.githubusercontent.com/45258028/223449483-ef30f99f-efae-4e89-b7b5-dcb5634b8a0f.png">

<img width="931" alt="image" src="https://user-images.githubusercontent.com/45258028/223449546-76ef90f0-392e-4d76-ba5a-80c2c7381ed0.png">

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'

================================
Printing MONAI config...
================================
MONAI version: 0+untagged.2538.g579fe65
Numpy version: 1.23.5
Pytorch version: 1.13.1
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 579fe65c96dce810ab06fcdbf37c142de8ef6815
MONAI __file__: /home/azureuser/localfiles/Repos/MONAI/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.11
ITK version: 5.3.0
Nibabel version: 5.0.1
scikit-image version: 0.20.0
Pillow version: 9.4.0
Tensorboard version: 2.12.0
gdown version: 4.6.4
TorchVision version: 0.14.1
tqdm version: 4.65.0
lmdb version: 1.4.0
psutil version: 5.9.4
pandas version: 1.5.3
einops version: 0.6.0
transformers version: 4.21.3
mlflow version: 2.2.1
pynrrd version: 1.0.0

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 20.04.5 LTS
Platform: Linux-5.15.0-1031-azure-x86_64-with-glibc2.31
Processor: x86_64
Machine: x86_64
Python version: 3.10.9
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='/home/azureuser/.vscode-server/data/logs/20230307T104704/remoteagent.log', fd=19, position=6400, mode='a', flags=33793), popenfile(path='/home/azureuser/.vscode-server/data/logs/20230307T104704/ptyhost.log', fd=20, position=0, mode='a', flags=33793)]
Num physical CPUs: 24
Num logical CPUs: 24
Num usable CPUs: 24
CPU usage (%): [3.7, 3.7, 3.7, 3.7, 3.7, 4.0, 3.7, 3.4, 3.4, 3.4, 3.4, 3.7, 3.4, 3.7, 3.4, 3.7, 3.4, 3.7, 5.7, 3.4, 3.4, 3.7, 4.0, 99.7]
CPU freq. (MHz): 2594
Load avg. in last 1, 5, 15 mins (%): [3.9, 9.4, 8.6]
Disk usage (%): 66.3
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 440.9
Available memory (GB): 433.0
Used memory (GB): 5.0

================================
Printing GPU config...
================================
Num GPUs: 4
Has CUDA: True
CUDA version: 11.6
cuDNN enabled: True
cuDNN version: 8302
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_61', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'compute_37']
GPU 0 Name: Tesla P100-PCIE-16GB
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 56
GPU 0 Total memory (GB): 15.9
GPU 0 CUDA capability (maj.min): 6.0
GPU 1 Name: Tesla P100-PCIE-16GB
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 56
GPU 1 Total memory (GB): 15.9
GPU 1 CUDA capability (maj.min): 6.0
GPU 2 Name: Tesla P100-PCIE-16GB
GPU 2 Is integrated: False
GPU 2 Is multi GPU board: False
GPU 2 Multi processor count: 56
GPU 2 Total memory (GB): 15.9
GPU 2 CUDA capability (maj.min): 6.0
GPU 3 Name: Tesla P100-PCIE-16GB
GPU 3 Is integrated: False
GPU 3 Is multi GPU board: False
GPU 3 Multi processor count: 56
GPU 3 Total memory (GB): 15.9
GPU 3 CUDA capability (maj.min): 6.0
```

**Additional context**
This test does not always fail. ~50% of the time, it passes, seeming to load the segresnet checkpoint instead for ensembling:

```shell
=> loaded checkpoint /tmp/tmpqmdgiqwz/workdir/segresnet_0/model/model.pt (epoch 1) (best_metric 0.4888312518596649)
Total parameters count 87164200 distributed False
monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
Inference complete, time 0.26s shape torch.Size([2, 24, 24, 24]) {'image': '/tmp/tmpqmdgiqwz/dataroot/val_001.fake.nii.gz'}
segresnet_0
.
----------------------------------------------------------------------
Ran 1 test in 202.620s

OK
```

**Comment by mingxin-zheng** [comment]
Hi @dongyang0122 , could you take a look at this issue?

**Comment by mingxin-zheng** [comment]
After some investigation with the help of @KumoLiu ,  the non-deterministic behavior comes from the usage of `nn.Upsample` in DiNTs and SegResNet2D templates.

**Comment by mingxin-zheng** [comment]
Hi @peterhessey , I tried this on a few different machines with different GPUs (ampere, pascal arch), but did not see it happens for me. My guess for the cause is that the Dints inferer loads a wrong config from the dummy training process, where some parameters are being optimized to fit the GPU model. Is this issue still bothering you?

**Comment by peterhessey** [comment]
Hi @mingxin-zheng, following from your comment I have updated to the latest `dev` branch, created a new python 3.8 environment (`conda create --name monai-test python=3.8  && conda activate monai-test && pip install -r requirements-dev.txt && pip install monai`) and tried running `python -m tests.test_auto3dseg_ensemble`. Unfortunately now my test is failing with a different error during the `dints_0/scripts/train.py` call:

```shell
2023-03-29 10:00:26,722 - INFO - Launching: torchrun --nnodes=1 --nproc_per_node=4 /tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py run --config_file='/tmp/tmpsjm9srip/workdir/dints_0/configs/hyper_parameters_search.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_validate.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_infer.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_train.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/hyper_parameters.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/network.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/network_search.yaml' --training#num_images_per_batch=2 --training#num_epochs=2 --training#num_epochs_per_validation=1 --training#determ=True
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
[info] number of GPUs: 4
2023-03-29 10:00:33,670 - Added key: store_based_barrier_key:1 to store for rank: 3
[info] number of GPUs: 4
2023-03-29 10:00:33,781 - Added key: store_based_barrier_key:1 to store for rank: 2
[info] number of GPUs: 4
2023-03-29 10:00:33,931 - Added key: store_based_barrier_key:1 to store for rank: 0
[info] number of GPUs: 4
2023-03-29 10:00:33,961 - Added key: store_based_barrier_key:1 to store for rank: 1
2023-03-29 10:00:33,961 - Rank 1: Completed store-based barrier for key:store_based_barrier_key:1 with 4 nodes.
[info] world_size: 4
train_files: 2
val_files: 1
2023-03-29 10:00:33,962 - Rank 0: Completed store-based barrier for key:store_based_barrier_key:1 with 4 nodes.
[info] world_size: 4
train_files: 2
val_files: 1
2023-03-29 10:00:33,964 - Rank 2: Completed store-based barrier for key:store_based_barrier_key:1 with 4 nodes.
[info] world_size: 4
2023-03-29 10:00:33,964 - Rank 3: Completed store-based barrier for key:store_based_barrier_key:1 with 4 nodes.
[info] world_size: 4
train_files: 2
val_files: 1
train_files: 2
val_files: 1
Traceback (most recent call last):
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 486, in <module>
    fire.Fire()
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 218, in run
    lr_scheduler_part = parser.get_parsed_content(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_parser.py", line 271, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 185, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 155, in _resolve_one_item
    self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 167, in _resolve_one_item
    item.evaluate(globals={f"{self._vars}": self.resolved_content}) if run_eval else item
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 354, in evaluate
    optional_module = self._parse_import_string(value[len(self.prefix) :])
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 324, in _parse_import_string
    node = first(ast.iter_child_nodes(ast.parse(import_string)))
  File "/anaconda/envs/monai-test/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    __local_refs['training#num_epochs'] // __local_refs['training#num_epochs']_per_validation + 1
                                                                              ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 486, in <module>
    fire.Fire()
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 218, in run
    lr_scheduler_part = parser.get_parsed_content(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_parser.py", line 271, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 185, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 155, in _resolve_one_item
    self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 167, in _resolve_one_item
    item.evaluate(globals={f"{self._vars}": self.resolved_content}) if run_eval else item
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 354, in evaluate
    optional_module = self._parse_import_string(value[len(self.prefix) :])
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 324, in _parse_import_string
    node = first(ast.iter_child_nodes(ast.parse(import_string)))
  File "/anaconda/envs/monai-test/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    __local_refs['training#num_epochs'] // __local_refs['training#num_epochs']_per_validation + 1
                                                                              ^
SyntaxError: invalid syntax
num_epochs 2
num_epochs_per_validation 1
Traceback (most recent call last):
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 486, in <module>
    fire.Fire()
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 218, in run
    lr_scheduler_part = parser.get_parsed_content(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_parser.py", line 271, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 185, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 155, in _resolve_one_item
    self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 167, in _resolve_one_item
    item.evaluate(globals={f"{self._vars}": self.resolved_content}) if run_eval else item
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 354, in evaluate
    optional_module = self._parse_import_string(value[len(self.prefix) :])
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 324, in _parse_import_string
    node = first(ast.iter_child_nodes(ast.parse(import_string)))
  File "/anaconda/envs/monai-test/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    __local_refs['training#num_epochs'] // __local_refs['training#num_epochs']_per_validation + 1
                                                                              ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 486, in <module>
    fire.Fire()
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py", line 218, in run
    lr_scheduler_part = parser.get_parsed_content(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_parser.py", line 271, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 185, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 155, in _resolve_one_item
    self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 167, in _resolve_one_item
    item.evaluate(globals={f"{self._vars}": self.resolved_content}) if run_eval else item
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 354, in evaluate
    optional_module = self._parse_import_string(value[len(self.prefix) :])
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/monai/bundle/config_item.py", line 324, in _parse_import_string
    node = first(ast.iter_child_nodes(ast.parse(import_string)))
  File "/anaconda/envs/monai-test/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    __local_refs['training#num_epochs'] // __local_refs['training#num_epochs']_per_validation + 1
                                                                              ^
SyntaxError: invalid syntax
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 39332) of binary: /anaconda/envs/monai-test/bin/python
Traceback (most recent call last):
  File "/anaconda/envs/monai-test/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/torch/distributed/run.py", line 794, in main
    run(args)
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/torch/distributed/run.py", line 785, in run
    elastic_launch(
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/anaconda/envs/monai-test/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 250, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-03-29_10:00:38
  host      : localhost
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 39333)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[2]:
  time      : 2023-03-29_10:00:38
  host      : localhost
  rank      : 2 (local_rank: 2)
  exitcode  : 1 (pid: 39334)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[3]:
  time      : 2023-03-29_10:00:38
  host      : localhost
  rank      : 3 (local_rank: 3)
  exitcode  : 1 (pid: 39335)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-03-29_10:00:38
  host      : localhost
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 39332)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
E
======================================================================
ERROR: test_ensemble (__main__.TestEnsembleBuilder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/azureuser/localfiles/Repos/MONAI_MONAI/tests/test_auto3dseg_ensemble.py", line 146, in test_ensemble
    algo.train(_train_param)
  File "/tmp/tmpsjm9srip/workdir/algorithm_templates/dints/scripts/algo.py", line 443, in train
  File "/home/azureuser/localfiles/Repos/MONAI_MONAI/monai/apps/auto3dseg/bundle_gen.py", line 193, in _run_cmd
    normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
  File "/anaconda/envs/monai-test/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['torchrun', '--nnodes=1', '--nproc_per_node=4', '/tmp/tmpsjm9srip/workdir/dints_0/scripts/train.py', 'run', "--config_file='/tmp/tmpsjm9srip/workdir/dints_0/configs/hyper_parameters_search.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_validate.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_infer.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/transforms_train.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/hyper_parameters.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/network.yaml','/tmp/tmpsjm9srip/workdir/dints_0/configs/network_search.yaml'", '--training#num_images_per_batch=2', '--training#num_epochs=2', '--training#num_epochs_per_validation=1', '--training#determ=True']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 1 test in 88.091s

```
 
Here's my `python -c 'import monai; monai.config.print_debug_info()'` now in case it's useful:

```shell
================================
Printing MONAI config...
================================
MONAI version: 1.2.0rc2+17.gbe3d1386
Numpy version: 1.23.5
Pytorch version: 2.0.0+cu117
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: be3d13869d9e0060d17a794d97d528d4e4dcc1fc
MONAI __file__: /home/azureuser/localfiles/Repos/MONAI_MONAI/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.11
ITK version: 5.3.0
Nibabel version: 5.0.1
scikit-image version: 0.20.0
Pillow version: 9.4.0
Tensorboard version: 2.12.0
gdown version: 4.7.1
TorchVision version: 0.15.1+cu117
tqdm version: 4.65.0
lmdb version: 1.4.0
psutil version: 5.9.4
pandas version: 1.5.3
einops version: 0.6.0
transformers version: 4.21.3
mlflow version: 2.2.2
pynrrd version: 1.0.0

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 20.04.5 LTS
Platform: Linux-5.15.0-1031-azure-x86_64-with-glibc2.17
Processor: x86_64
Machine: x86_64
Python version: 3.8.16
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='/home/azureuser/.vscode-server/data/logs/20230329T085033/remoteagent.log', fd=19, position=3135, mode='a', flags=33793), popenfile(path='/home/azureuser/.vscode-server/data/logs/20230329T085033/ptyhost.log', fd=20, position=0, mode='a', flags=33793)]
Num physical CPUs: 24
Num logical CPUs: 24
Num usable CPUs: 24
CPU usage (%): [0.7, 3.3, 4.3, 3.6, 3.6, 3.3, 3.3, 3.6, 3.3, 3.3, 0.3, 1.6, 3.6, 3.6, 3.6, 3.6, 5.3, 3.3, 3.3, 3.9, 3.6, 3.9, 3.6, 100.0]
CPU freq. (MHz): 2594
Load avg. in last 1, 5, 15 mins (%): [4.0, 5.9, 6.0]
Disk usage (%): 75.8
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 440.9
Available memory (GB): 432.4
Used memory (GB): 5.6

================================
Printing GPU config...
================================
Num GPUs: 4
Has CUDA: True
CUDA version: 11.7
cuDNN enabled: True
cuDNN version: 8500
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80', 'sm_86']
GPU 0 Name: Tesla P100-PCIE-16GB
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 56
GPU 0 Total memory (GB): 15.9
GPU 0 CUDA capability (maj.min): 6.0
GPU 1 Name: Tesla P100-PCIE-16GB
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 56
GPU 1 Total memory (GB): 15.9
GPU 1 CUDA capability (maj.min): 6.0
GPU 2 Name: Tesla P100-PCIE-16GB
GPU 2 Is integrated: False
GPU 2 Is multi GPU board: False
GPU 2 Multi processor count: 56
GPU 2 Total memory (GB): 15.9
GPU 2 CUDA capability (maj.min): 6.0
GPU 3 Name: Tesla P100-PCIE-16GB
GPU 3 Is integrated: False
GPU 3 Is multi GPU board: False
GPU 3 Multi processor count: 56
GPU 3 Total memory (GB): 15.9
GPU 3 CUDA capability (maj.min): 6.0
```

Should I open a separate issue for this? As it seems to be a different error to the one we were seeing previously, and is failing 100% of the time.

**Comment by wyli** [comment]
the last different issue should have been fixed by https://github.com/Project-MONAI/MONAI/commit/a77de4adff49037ffb24ea2cae217fac40302181 https://github.com/Project-MONAI/MONAI/issues/5931

closing this issue for now, please feel free to create new tickets for the same.
